_posts/2021-09-08-deep-dive-into-a-debezium-community-connector-scylla-cdc-source-connector.adoc

```diff
@@ -0,0 +1,229 @@
+----
+
+Finally, let’s see the contents of the modified table:
+
+[source,sql]
+----
+SELECT * FROM ks.orders;
+
+ user | order_id | order_name
+------+----------+-------------
+  Tim |        1 |   pineapple
+Alice |        2 | blueberries
+----
+
+Looking only at this table, you cannot easily reconstruct all modifications that took place, e.g. Tim’s order name before the UPDATE. Let’s look at the CDC log, easily accessible as a table (some columns truncated for clarity):
+
+[source,sql]
+----
+SELECT * FROM ks.orders_scylla_cdc_log;
+
+ cdc$stream_id | cdc$time |  | cdc$operation | order_id | order_name  |     user
+---------------+----------+-...-+---------------+----------+-------------+----------
+    0x2e46a... |  7604... | ... |             2 |        1 |       apple |      Tim
+    0x2e46a... |  8fdc... | ... |             1 |        1 |   pineapple |      Tim
+    0x41400... |  808e... | ... |             2 |        2 | blueberries |    Alice
+----
+
+All three operations are visible in the CDC log. There are two INSERTs (`cdc$operation = 2`) and one UPDATE (`cdc$operation = 1`). For each operation, its timestamp is also preserved (`cdc$time` column as time UUID).
+
+== Choosing Debezium
+
+As shown in the previous section, Scylla CDC can be easily queried as a regular table. To get the latest operations in real-time, you poll the table with appropriate time ranges. To make this easier, we developed client libraries for https://github.com/scylladb/scylla-cdc-java[Java] and https://github.com/scylladb/scylla-cdc-go[Go].
+
+When we thought about how our customers could access the CDC log, using it with Kafka seemed like the most accessible method. Therefore, we decided to develop a source connector for Scylla CDC.
+
+For our first proof-of-concept, we implemented the source connector using the https://kafka.apache.org/documentation.html#connect_development[Kafka Connect API]. This prototype was crucial for us to determine if the connector could scale horizontally (described later in this blogpost). 
+
+However, we quickly realized that by using only the Kafka Connect API, we would have to reimplement a lot of functionality already present in other connectors. We also wanted our connector to be a good citizen in the Kafka community, sticking to the best practices and conventions. That’s exactly why we choose Debezium! 
+
+Thanks to this, when you start the Scylla CDC Source Connector, the configuration parameters will be immediately familiar, as many of them are common with other Debezium connectors. The generated data change events have the same https://javadoc.io/static/io.debezium/debezium-core/0.7.2/io/debezium/data/Envelope.html[Envelope] structure as generated by other Debezium connectors. This similarity allows for the use of many standard Debezium features, such as https://debezium.io/documentation/reference/1.6/transformations/event-flattening.html[New Record State Extraction] [.underline]#.#
```

Comment:
```suggestion
Thanks to this, when you start the Scylla CDC Source Connector, the configuration parameters will be immediately familiar, as many of them are common with other Debezium connectors. The generated data change events have the same https://javadoc.io/static/io.debezium/debezium-core/0.7.2/io/debezium/data/Envelope.html[Envelope] structure as generated by other Debezium connectors. This similarity allows for the use of many standard Debezium features, such as link:/documentation/reference/1.6/transformations/event-flattening.html[New Record State Extraction] [.underline]#.#
```

That way, it's a relative link, also working correctly with local website builds, etc.

Reply:
What is the underline part at the end about?